c2rust-transpile/src/lib.rs

```diff
@@ -227,10 +227,65 @@ fn get_module_name(
 
 /// Main entry point to transpiler. Called from CLI tools with the result of
 /// clap::App::get_matches().
-pub fn transpile(tcfg: TranspilerConfig, cc_db: &Path, extra_clang_args: &[&str]) {
+pub fn transpile(tcfg: TranspilerConfig, source_or_cc_db: &Path, extra_clang_args: &[&str]) {
     diagnostics::init(tcfg.enabled_warnings.clone(), tcfg.log_level);
 
-    let lcmds = get_compile_commands(cc_db, &tcfg.filter).unwrap_or_else(|_| {
+    let build_dir = get_build_dir(&tcfg, source_or_cc_db);
```

Comment:
rather than adding a bunch of code to `transpile`, might it make sense to extract this into its own function? If not, a comment or two might be in order.